examples/calculations/submit_singlepoint.py

```diff
@@ -0,0 +1,167 @@
+    return model
+
+
+def load_structure(value: Union[str, Path, int, None]) -> StructureData:
+    """
+    Load a StructureData instance from the given input.
+
+    The input can be either a path to a structure file, a node PK (int),
+    or None. If the input is None, a default StructureData instance for NaCl
+    with a rocksalt structure will be created.
+
+    Parameters
+    ----------
+    value : str, int or None
+        The input value representing either a path to a structure file, a node PK,
+        or None.
+
+    Returns
+    -------
+    StructureData
+        The loaded or created StructureData instance.
+
+    Raises
+    ------
+    click.BadParameter
+        If the input is not a valid path to a structure file or a node PK.
+    """
+    if value is None:
+        structure = StructureData(ase=bulk("NaCl", "rocksalt", 5.63))
+    elif isinstance(value, int) or (isinstance(value, str) and value.isdigit()):
+        structure_pk = int(value)
+        structure = load_node(structure_pk)
+    elif Path.exists(value):
+        structure = StructureData(ase=read(value))
+    else:
+        raise click.BadParameter(
+            f"Invalid input: {value}. Must be either node PK (int) or a valid \
+                path to a structure file."
+        )
+    return structure
+
+
+def singlepoint(params: dict):
+    """
+    Prepare inputs and run a single point calculation.
+
+    Parameters
+    ----------
+    params : dict
+        A dictionary containing the input parameters for the calculations
+
+    Returns
+    -------
+    tuple
+        A tuple containing the exit code and the node of the calculation.
+    """
+    structure = load_structure(params["file"])
+
+    # Select model to use
+    model = load_model(params["model"], params["architecture"])
+
+    # Select calculation to use
+    Singlepointcalc = CalculationFactory("janus.sp")
+
+    # Define inputs
+    inputs = {
+        "metadata": {"options": {"resources": {"num_machines": 1}}},
+        "code": params["code"],
+        "architecture": Str(params["architecture"]),
+        "structure": structure,
+        "calctype": Str(params["calctype"]),
+        "model": model,
+        "precision": Str(params["precision"]),
+        "device": Str(params["device"]),
+    }
+
+    # Run calculation
+    result, node = run_get_node(Singlepointcalc, **inputs)
+    print(f"Printing results from calculation: {result}")
+    print(f"Printing node of calculation: {node}")
+
+
+# Arguments and options to give to the cli when running the script
+@click.command("cli")
+@click.argument("codelabel", type=str)
+@click.option("--calctype", default="singlepoint", type=str)
+@click.option(
+    "--file",
+    default=None,
+    type=Union[str, Path, int],
+    help="Specify the structure (aiida node or path to a structure file)",
+)
+@click.option(
+    "--model", default=None, type=str, help="Specify path or url of the model to use"
```

Comment:
```suggestion
    "--model", default=None, type=Union[str, Path], help="Specify path or url of the model to use"
```
Not sure how it handles a path if it can be either, but probably doesn't hurt?